components/SnackBar/SnackBar.jsx

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import styled from 'styled-components';
+import PropTypes, { oneOf } from 'prop-types';
+import Colors from '../Colors';
+import Button from '../Button';
+import { Row, Col } from '../Grid';
+import uniqId from '../shared/uniqId';
+
+const a11yFocusTab = `outline: auto;`;
+
+const getBackgroundBySkin = skin => {
+  const color = skin.toUpperCase();
+  const indexColor = skin === 'cobalt' ? '500' : '700';
+  return `
+      background-color: ${Colors[color][indexColor]}
+  `;
+};
+
+const SnackBarDialog = styled.div`
+  align-items: center;
+  border-radius: 4px;
+  color: ${Colors.WHITE};
+  display: flex;
+  justify-content: space-between;
+  min-height: 48px;
+  padding: 13px 16px;
+
+  ${({ skin }) => getBackgroundBySkin(skin)}
+`;
+
+const TextContainer = styled.strong`
+  font-weight: normal;
+  padding-right: 8px;
+
+  :focus {
+    ${a11yFocusTab}
+  }
+`;
+
+const CloseIcon = styled(Button.Icon).attrs({
+  icon: 'close',
+})`
+  color: ${Colors.WHITE};
+
+  :hover,
+  :focus {
+    background-color: transparent;
+    box-shadow: none;
+    color: ${Colors.WHITE};
+    ${a11yFocusTab}
+  }
+`;
+
+const ActionButton = styled(Button)`
+  color: ${Colors.WHITE};
+  font-weight: bold;
+  text-transform: uppercase;
+  white-space: nowrap;
+
+  :hover,
+  :focus {
+    text-decoration: none;
+    color: ${Colors.WHITE};
+  }
+
+  :focus {
+    ${a11yFocusTab}
+  }
+`;
+
+const ActionsSection = styled.div`
+  align-items: center;
+  display: flex;
+`;
+
+const DialogBlock = styled.section`
+  bottom: 0;
+  left: 0;
+  position: fixed;
+  width: 100%;
+`;
+
+class SnackBar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.snackBarSection = document.createElement('section');
+  }
+
+  componentDidMount() {
+    const { body } = document;
+    body.appendChild(this.snackBarSection);
+
+    this.objectFocus(document.querySelector('.action-button'));
+
+    const { onClose, secondsToClose } = this.props;
+
+    setTimeout(() => {
```

Comment:
I think you must save this `setTimeout` to a class property to "clear" this on componentWillUnmount();

If you open a snackbar and count to 3, click on close icon, open another snackbar and wait, instead of closing in 6 seconds the new one will close after 3 seconds.